pkg/inclusion/paths.go

```diff
@@ -32,8 +33,11 @@ func calculateCommitPaths(squareSize, start, msgShareLen int) []path {
 		if i == endRow {
 			end = normalizedEndIndex
 		}
-		coord := calculateSubTreeRootCoordinates(maxDepth, start, end)
-		for _, c := range coord {
+
+		subTreeRootMaxHeight := int(math.Log2(float64(types.MinSquareSize(uint64(msgShareLen)))))
```

Comment:
Agreed. We may be able to avoid one type conversions if `MinSquareSize` accepts an `int` arg which can be done by adopting generics so will do shortly. I don't immediately see an alternative to avoid the remaining type conversions but open to ideas.

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/937/commits/671bc800acecfdeef426db580811fc0baeeda037